envoy/Makefile

```diff
@@ -79,24 +98,28 @@ all: clean-bins release
 else
 BAZEL_BUILD_OPTS =
 
-all: clean-bins envoy $(GO_TARGETS)
+all: $(ENVOY_BIN) $(GO_TARGETS)
 endif
 
 debug: envoy-debug $(GO_TARGETS)
 
 release: envoy-release $(GO_TARGETS)
 
-envoy: force-non-root
+$(ENVOY_BIN) envoy: $(CILIUM_ENVOY_SOURCES) $(CILIUM_API_SOURCES)
+	make force-non-root
 	$(BAZEL) $(BAZEL_OPTS) build $(BAZEL_BUILD_OPTS) //:envoy
+	touch $(ENVOY_BIN)
 
 # Allow root build for release
-$(ENVOY_BIN) envoy-release: force
+envoy-release: force
 	$(BAZEL) $(BAZEL_OPTS) build $(BAZEL_BUILD_OPTS) -c opt //:envoy
 
-envoy-debug: force-non-root
+envoy-debug: $(CILIUM_ENVOY_SOURCES) $(CILIUM_API_SOURCES)
+	make force-non-root
 	$(BAZEL) $(BAZEL_OPTS) build $(BAZEL_BUILD_OPTS) -c dbg //:envoy
 
-$(CHECK_FORMAT): force-non-root
+$(CHECK_FORMAT): $(CILIUM_ENVOY_SOURCES) $(CILIUM_API_SOURCES)
```

Comment:
This makes sure that the check_format.py is made available, and does not depend on any of our sources so IMO this change should be reverted.